docs/curate-text/process-data/language-management/index.md

```diff
@@ -12,57 +12,211 @@ modality: "text-only"
+
+* **Data Format**: JSONL (JSON Lines) input with text content
+  - Default field name: `text`
+  - Custom field support: Specify with `text_field` parameter
+
+* **Cluster Setup** (if applicable):
+  - Ensure FastText model file is accessible to all workers
+  - Use shared filesystem, network storage, or object storage (S3, GCS, etc.)
+
+## Basic Language Filtering
+
+### Quick Start Example
+
+Filter documents by language using FastText:
 
 ```python
 from nemo_curator.pipeline import Pipeline
 from nemo_curator.stages.text.io.reader import JsonlReader
+from nemo_curator.stages.text.io.writer import JsonlWriter
 from nemo_curator.stages.text.modules import ScoreFilter
 from nemo_curator.stages.text.filters import FastTextLangId
 
-# 1) Build the pipeline
-pipeline = Pipeline(name="language_management")
+# Create language filtering pipeline
+pipeline = Pipeline(name="language_filtering")
 
-# Read JSONL files into document batches
+# 1. Read JSONL input files
 pipeline.add_stage(
-    JsonlReader(file_paths="input_data/*.jsonl", files_per_partition=2)
+    JsonlReader(
+        file_paths="input_data/",
+        files_per_partition=2  # Process 2 files per partition
+    )
 )
 
-# Identify languages and keep docs above a confidence threshold
+# 2. Identify languages and filter by confidence threshold
 pipeline.add_stage(
     ScoreFilter(
-        FastTextLangId(model_path="/path/to/lid.176.bin", min_langid_score=0.3),
-        score_field="language",
+        FastTextLangId(
+            model_path="/path/to/lid.176.bin",  # Path to FastText model
+            min_langid_score=0.3                # Minimum confidence (0.0-1.0)
+        ),
+        score_field="language"  # Output field for language code
     )
 )
 
-# 2) Execute
+# 3. Write filtered results
+pipeline.add_stage(
+    JsonlWriter(path="output_filtered/")
+)
+
+# Execute pipeline (uses XennaExecutor by default)
 results = pipeline.run()
 ```
+**Parameters explained:**
+- `model_path`: Absolute path to FastText model file (`lid.176.bin` or `lid.176.ftz`)
+- `min_langid_score`: Minimum confidence score (0.0 to 1.0). Documents below this threshold are filtered out
+- `score_field`: Field name to store detected language code (e.g., "en", "es", "zh")
+- `files_per_partition`: Number of files to process per partition (tune based on file sizes)
 
----
+**Output format:**
+Each document will include a `language` field with the detected language code:
+```json
+{"text": "This is an English document.", "language": "en"}
+{"text": "Este es un documento en español.", "language": "es"}
+```
+## Integration with HTML Extraction
 
-## Language Processing Capabilities
+When processing HTML content (e.g., Common Crawl), CLD2 provides language hints automatically:
+
+```python
+from nemo_curator.stages.text.download import CommonCrawlDownloadExtractStage
+
+# HTML extraction automatically uses CLD2 for language hints
+pipeline.add_stage(CommonCrawlWarcDownloader(...))
```

Comment:
Imports `CommonCrawlDownloadExtractStage` but tries to use `CommonCrawlWarcDownloader`.